components/dashboard/src/org-admin/MaintenanceNotificationCard.tsx

```diff
@@ -0,0 +1,144 @@
+                    onClick={toggleNotification}
+                    disabled={isLoading}
+                >
+                    {isLoading ? "Loading..." : isNotificationEnabled ? "Disable" : "Enable"}
+                </Button>
+            </div>
+
+            {/* Message input section */}
+            <div className="mt-4">
+                <label
+                    htmlFor="maintenance-message"
+                    className="block text-sm font-medium text-gray-700 dark:text-gray-300 mb-2"
+                >
+                    Notification Message
+                </label>
+                {isEditing ? (
+                    <div>
+                        <textarea
+                            id="maintenance-message"
+                            className="w-full px-3 py-2 border border-gray-300 dark:border-gray-600 rounded-md shadow-sm focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 dark:bg-gray-700 dark:text-white"
+                            rows={3}
+                            value={message}
+                            onChange={(e) => setMessage(e.target.value)}
+                            placeholder="Enter a message to display in the notification banner"
+                        />
+                        <div className="mt-2 flex justify-end space-x-2">
+                            <Button
+                                variant="secondary"
+                                onClick={() => {
+                                    setMessage(notificationMessage || DEFAULT_MESSAGE);
+                                    setIsEditing(false);
+                                }}
+                            >
+                                Cancel
+                            </Button>
+                            <Button onClick={saveMessage}>Save</Button>
+                        </div>
+                    </div>
+                ) : (
+                    <div>
+                        <div className="px-3 py-2 border border-gray-300 dark:border-gray-600 rounded-md bg-gray-50 dark:bg-gray-700 text-gray-700 dark:text-gray-300 min-h-[4rem]">
+                            {message}
+                        </div>
+                        <div className="mt-2 flex justify-end">
+                            <Button variant="secondary" onClick={() => setIsEditing(true)}>
+                                Edit Message
+                            </Button>
+                        </div>
+                    </div>
+                )}
+            </div>
+
+            <div className="mt-4">
+                <label className="block text-sm font-semibold text-gray-700 dark:text-gray-300 mb-2">Preview</label>
+                <Alert type="warning" className="mb-0">
+                    <div className="flex items-center">
+                        <span className="font-semibold">Scheduled Maintenance:</span>
+                        <span className="ml-2">{message}</span>
+                    </div>
```

Comment:
To also fix wrapping here on narrow viewports:
```suggestion
                    <div className="flex items-center flex-wrap gap-2">
                        <span className="font-semibold">Scheduled Maintenance:</span>
                        <span>{message}</span>
                    </div>
```